packages/client/src/forms/register/mappings/mutation/field-mappings.ts

```diff
@@ -169,48 +192,93 @@ export const fieldToAddressLineTransformer =
     }
     let address: IAddress | undefined = (
       sectionData.address as [{ type: string; line: IFormFieldValue[] }]
-    ).find((addr) => addr.type === addressType)
+    ).find((addr) => addr.type === config.useCase)
     if (!address) {
       address = {
-        type: addressType,
-        line: ['', '', '', '', '', '', '', '', '', '', '', '', '', '', ''] // lines must be available as empty strings for GraphQL to parse all options
+        type: config.useCase,
+        line: getDefaultAddressLines()
       }
       sectionData.address.push(address)
     }
 
-    address.line[lineNumber] = `${draftData[sectionId][field.name]}`
+    if (config.lineNumber || config.lineNumber === 0) {
+      address.line[config.lineNumber] = `${draftData[sectionId][field.name]}`
+    } else if (config.transformedFieldName) {
+      address[
+        !config.transformedFieldName ? field.name : config.transformedFieldName
+      ] = `${draftData[sectionId][field.name]}`
+    }
```

Comment:
We might want to handle the case when `config.transformedFieldName` is not provided

Reply:
As an FYI. ... It is not provided when lineNumber is sent [here](https://github.com/opencrvs/opencrvs-farajaland/blob/af4b3be97aae96d36027a5c9131d59d5b00cbc7d/src/utils/address-utils.ts#L741).  I think all the mapping functions could be improved with better error handling. I will hopefully get around to it later.